gtda/externals/python/ripser_interface.py

```diff
@@ -162,6 +162,19 @@ def _resolve_symmetry_conflicts(coo):
         return _row, _col, _data
 
 
+def _collapse_coo(row, col, data, thresh):
+    """Run edge collapser on off-diagonal data and then reinsert diagonal
+    data."""
```

Comment:
I don't know, I understand perfectly the comment. But now, data is processed as if it all inputs that will be collapsed could contain non 0 diagonals entries.

And if the user did not build correctly it's input (I got in the past someone with non zero entries in the diagonal, and it wasn't intended) this will run without a problem, but detecting that the output could not be as expected or worse they don't know that their input data contains no "normal" entries (I really don't know how to call them). Will become really hard to spot.

In my opinion, in order to accept this, the user should maybe by an input flag, allow explicitly to process non 0 diagonal entries.

What do you think @ulupo ?

Maybe there's already a check like that somewhere else in the library that I'm not aware of ?

Reply:
I don't understand this comment, sorry :(. The way I see it, if all the diagonal entries are zero then the row_diag, col_diag and data_diag in the body of the function will be empty and things go as they always did. If the diagonal entries are not zero then we want to take them seriously and not assume it was a mistake.

Reply:
You're right, sorry, we clearly support non-zeros entries on the diagonal. I don't know why, but in my head I was thinking about `flagser`. 

Sorry again discard my comment, all good !

Reply:
Thanks for the review!!